smssync/src/main/java/org/addhen/smssync/presentation/service/ServiceControl.java

```diff
@@ -207,13 +207,8 @@ public boolean isConnected() {
         return Utility.isConnected(mContext);
     }
 
-    public enum SchedulerInstance {
-        INSTANCE;
-
-        private Scheduler getScheduler(Context context, FileManager fileManager, Intent intent,
-                int requestCode) {
-            return new Scheduler(context, fileManager, intent, requestCode,
-                    PendingIntent.FLAG_UPDATE_CURRENT);
-        }
+    private Scheduler getScheduler(Intent intent, int requestCode) {
```

Comment:
Would it make more sense for `Scheduler` itself to enforce the singletonism?  Any use of `Scheduler` outside `ServiceControl` can quite easily create new instances.  In fact, that's happening in https://github.com/ushahidi/SMSSync/blob/b8cd41ae83297151d5a3cc29ce52a9e328b6591d/smssync/src/main/java/org/addhen/smssync/presentation/receiver/BootReceiver.java#L82